backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/LibvirtVmXmlBuilder.java

```diff
@@ -2437,21 +2437,22 @@ private void writeDiskSource(VmDevice device, Disk disk, String dev, DiskVmEleme
         case MANAGED_BLOCK_STORAGE:
             ManagedBlockStorageDisk managedBlockStorageDisk = (ManagedBlockStorageDisk) disk;
             Map<String, String> metadata = new HashMap<>();
-            String path = (String) managedBlockStorageDisk.getDevice().get(DeviceInfoReturn.PATH);
+            String path;
+            if (Version.v4_7.lessOrEquals(vm.getCompatibilityVersion())) {
+                path = (String) managedBlockStorageDisk.getDevice().get(DeviceInfoReturn.MANAGED_PATH);
+            } else {
+                path = (String) managedBlockStorageDisk.getDevice().get(DeviceInfoReturn.PATH);
+            }
 
             if (managedBlockStorageDisk.getCinderVolumeDriver() == CinderVolumeDriver.RBD) {
-                // For rbd we need to pass the entire path since we rely on more than a single
-                // variable e.g: /dev/rbd/<pool-name>/<vol-name>
-                metadata = Collections.singletonMap("RBD", path);
+                metadata.put("RBD", path);
```

Comment:
So before we started with empty map, and then replaced it with Collections.singletonMap?
Using metadata looks good, but is this change needed to support managed_path or a cleanup that we can do later?

Reply:
unfortunately it is not a cleanup, singletonMap was incorrect because it's immutable which would prevent setting  metadata.put("managed", "true"); later on